http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/KubernetesServiceDbQueries.scala

```diff
@@ -0,0 +1,256 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import org.broadinstitute.dsde.workbench.leonardo.db.kubernetesClusterQuery.{unmarshalKubernetesCluster}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+import scala.concurrent.ExecutionContext
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.api._
+import cats.data.Chain
+import LeoProfile.mappedColumnImplicits._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.{dummyDate}
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.leonardo.db.nodepoolQuery.unmarshalNodepool
+
+object KubernetesServiceDbQueries {
+  def listFullApps(googleProject: Option[GoogleProject],
```

Comment:
its analogous to the word in our cluster queries, where 'full' does all joins possible. There is a 'minimal' query that only does a few joins, specified in a comment above

Reply:
I think it helps, because it makes the caller aware of what they are going to be getting back, and consider if the alternative lower-cost query is desirable. I'll add some comments. 

The alternative to this is specifying two different ADTs that have 1 field different (one having a list of apps, and the other not), which I think would be more confusing since it would require different handling at call site.